core/src/main/scala/kafka/log/LogCleanerManager.scala

```diff
@@ -64,6 +64,19 @@ private[log] class LogCleanerManager(val logDirs: Array[File], val logs: Pool[To
   @volatile private var dirtiestLogCleanableRatio = 0.0
   newGauge("max-dirty-percent", new Gauge[Int] { def value = (100 * dirtiestLogCleanableRatio).toInt })
 
+  /* a gauge for tracking the number of logCleaner runs */
+  @volatile private var numLogCleanRuns = 0
+  newGauge("num-runs", new Gauge[Int] { def value = numLogCleanRuns })
+
+  /* a gauge for tracking the last time of logCleaner run */
+  @volatile private var logCleanLastRun : Long = System.currentTimeMillis
+  newGauge("last-run-time", new Gauge[Long] { def value = logCleanLastRun })
+
+  /* a gauge for tracking the number of filthy logs. A non zero value indicates
```

Comment:
If the rate of incoming data is higher than the ability of the log cleaner to clean it, it's possible that num-filthy-logs will stay above zero, even though the log cleaner is making progress. Seems unlikely, though, that this would happen.

Not sure how to phrase it, but I think the operator should expect that num-filthy-logs returns to 0 in a reasonable amount of time. If it doesn't, then either the thread has died, or the threads aren't keeping up with the ingest rate (which might mean you need more threads).